pkg/pgmodel/metrics/database/metrics.go

```diff
@@ -92,6 +92,90 @@ var metrics = []metricQueryWrap{
 			},
 		),
 		query: `select count(*)::bigint from _timescaledb_catalog.chunk where dropped=false and compressed_chunk_id is not null`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "chunks_metrics_expired_count",
+				Help:      "The number of metrics chunks soon to be removed by maintenance jobs.",
+			},
+		),
+		query: `WITH conf AS MATERIALIZED (SELECT _prom_catalog.get_default_retention_period() AS def_retention)
+		SELECT count(*)::BIGINT
+		FROM _timescaledb_catalog.dimension_slice ds
+			 INNER JOIN _timescaledb_catalog.dimension d ON (d.id = ds.dimension_id)
+			 INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = d.hypertable_id)
+			 INNER JOIN _prom_catalog.metric m ON (m.table_name = h.table_name AND m.table_schema = h.schema_name)
+			 JOIN conf ON TRUE
+		WHERE ds.range_start < _timescaledb_internal.time_to_internal(now() - coalesce(m.retention_period, conf.def_retention))
+		  AND ds.range_end < _timescaledb_internal.time_to_internal(now() - coalesce(m.retention_period, conf.def_retention))`,
+	}, {
+		metric: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: util.PromNamespace,
+				Subsystem: "sql_database",
+				Name:      "chunks_metrics_uncompressed_count",
+				Help:      "The number of metrics chunks soon to be compressed by maintenance jobs.",
+			},
+		),
+		query: `WITH chunk_candidates AS MATERIALIZED (
+		    SELECT chcons.dimension_slice_id, h.table_name, h.schema_name
+		    FROM _timescaledb_catalog.chunk_constraint chcons
+			     INNER JOIN _timescaledb_catalog.chunk c ON c.id = chcons.chunk_id
+			     INNER JOIN _timescaledb_catalog.hypertable h ON h.id = c.hypertable_id
+		    WHERE c.dropped IS FALSE
+		      AND h.compression_state = 1 -- compression_enabled = TRUE
+		      AND (c.status & 1) != 1) -- only check for uncompressed chunks
+		SELECT count(*)::BIGINT
+		FROM chunk_candidates cc
+			 INNER JOIN _timescaledb_catalog.dimension_slice ds ON ds.id = cc.dimension_slice_id
+			 INNER JOIN _prom_catalog.metric m ON (m.table_name = cc.table_name AND m.table_schema = cc.schema_name)
+		WHERE (m.delay_compression_until IS NULL OR m.delay_compression_until < now())
```

Comment:
we should probably add a metric for counting chunks that would be compressed but are delayed as well. That's useful for debugging cases where backfill has caused disk increase due to delay in compression

Reply:
also applies to traces

Reply:
I've added it for the metrics data, but I couldn't find it for traces.